api/src/main/java/org/openmrs/module/metrics/api/db/hibernate/HibernateMetricsDAO.java

```diff
@@ -24,4 +31,41 @@ public MetricEvent saveMetricEvent(MetricEvent metricEvent) {
 		sessionFactory.getCurrentSession().saveOrUpdate(metricEvent);
 		return metricEvent;
 	}
+	
+	@Override
+	public Integer getEncounterObjectsByGivenDateRangeAndType(LocalDateTime startRange, LocalDateTime endRange,
+	        String encounterType) {
+		String sql = "SELECT count(*)"
```

Comment:
Please use parameterised queries instead of string concatenation to create SQL queries. Doing this opens us up to SQL Injection vulnerabilities and makes us dependent on `LocalDateTime#toString()` returning the right format.

```java
//@formatter:off
String sql = "select count(*) " +
"from event_records e " +
    "inner join encounter b on e.object_uuid = b.uuid " +
    "inner join encounter_type c on b.encounter_type = c.encounter_type_id " +
"where e.title = 'Encounter' and (e.date_created between :startRange and :endRange) and e.tags = 'CREATED' and c.name = :encounterType";

return (Integer) sessionFactory.getCurrentSession().createSQLQuery(sql)
    .setDate("startRange", Date.from(startRange.atZone(ZoneId.systemDefault()).toInstant())
    .setDate("endRange", Date.from(startRange.atZone(ZoneId.systemDefault()).toInstant())
    .setString("encounterType", encounterType).uniqueResult();
//@formatter:on
```